lib/pbench/agent/tool_meister.py

```diff
@@ -376,49 +362,41 @@ class PcpTransientTool(Tool):
 
     def __init__(self, name, tool_opts, **kwargs):
         super().__init__(name, tool_opts, **kwargs)
-        self.pmcd_args = None
+        self.pmcd_path = None
         self.pmcd_process = None
-        self.pmlogger_args = None
+        self.pmlogger_path = None
         self.pmlogger_process = None
-        if "/usr/libexec/pcp/bin" not in os.environ["PATH"]:
-            # FIXME - Shouldn't this be provided by the environment?
-            os.environ["PATH"] += f"{os.pathsep}/usr/libexec/pcp/bin"
-        self.pmcd_path = find_executable("pmcd")
-        self.pmlogger_path = find_executable("pmlogger")
 
     def install(self):
+        self.pmcd_path = find_executable("pmcd")
         if not self.pmcd_path:
             return (1, "pcp tool (pmcd) not found")
+        self.pmlogger_path = find_executable("pmlogger")
         if not self.pmlogger_path:
             return (1, "pcp tool (pmlogger) not found")
         return (0, "pcp tool (pmcd and pmlogger) properly installed")
 
-    def start(self):
-        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
-        if not self.tool_dir.is_dir():
-            raise RuntimeError(f"tool directory does not exist: {self.tool_dir}")
-        if not self.pmcd_path:
-            raise RuntimeError("Path to pmcd not provided")
-        if not self.pmlogger_path:
-            raise RuntimeError("Path to pmlogger not provided")
-        if self.pmcd_process is not None:
-            raise ToolException(
-                f"Tool({self.name}) has an unexpected pmcd process running"
-            )
-        if self.pmlogger_process is not None:
-            raise ToolException(
-                f"Tool({self.name}) has an unexpected pmlogger process running"
-            )
-
-        tool_dir = self.tool_dir / self.name.replace("-transient", "")
-        self.pmcd_args = [
+    def start(self, tool_dir: Path):
+        assert tool_dir is not None, "Logic bomb!  no tool directory provided!"
+        assert tool_dir.is_dir(), f"tool directory does not exist: {tool_dir}"
+        assert self.pmcd_path is not None, "Path to pmcd not provided"
+        assert self.pmlogger_path is not None, "Path to pmlogger not provided"
```

Comment:
These are more on the borderline, for me. This method expects that the tool paths were discovered previously, and there's an object state issue if they're not. However it's not quite clear cut as the `start` method's relationship to `install` is under control of the external caller, *not* of the internal class state. This means that the actual state is ambiguous. Maybe the class messed up and something cleared them between `install` and here, which would be deserving of an `assert` ... but maybe it just means the caller mis-used the class and failed to call `install`. That's a runtime error that can/should be fixed by the caller; it should be checked and reported in a production (`-O`) build because it's "undesirable" but not by any means "impossible" (or even "unlikely'), and it should be reported by a distinct exception that could be caught and handled.